itests/kit/node_unmanaged.go

```diff
@@ -24,3 +59,848 @@ type TestUnmanagedMiner struct {
+
+	r, err = tm.submitMessage(ctx, &miner14.ProveCommitSectors3Params{
+		SectorActivations:        []miner14.SectorActivationManifest{{SectorNumber: sectorNumber}},
+		SectorProofs:             [][]byte{sectorProof},
+		RequireActivationSuccess: true,
+	}, 0, builtin.MethodsMiner.ProveCommitSectors3)
+	req.NoError(err)
+	req.True(r.Receipt.ExitCode.IsSuccess())
+
+	tm.proofType[sectorNumber] = proofType
+
+	respCh := make(chan WindowPostResp, 1)
+
+	go tm.wdPostLoop(ctx, sectorNumber, respCh, true)
+
+	return sectorNumber, respCh
+}
+
+func (tm *TestUnmanagedMiner) OnboardCCSectorWithRealProofs(ctx context.Context, proofType abi.RegisteredSealProof) (abi.SectorNumber, chan WindowPostResp) {
+	req := require.New(tm.t)
+	sectorNumber := tm.currentSectorNum
+	tm.currentSectorNum++
+
+	// --------------------Create pre-commit for the CC sector -> we'll just pre-commit `sector size` worth of 0s for this CC sector
+
+	// Step 1: Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
+	preCommitSealRand := tm.waitPreCommitSealRandomness(ctx, sectorNumber)
+
+	// Step 2: Write empty bytes that we want to seal i.e. create our CC sector
+	tm.makeAndSaveCCSector(ctx, sectorNumber)
+
+	// Step 3: Generate a Pre-Commit for the CC sector -> this persists the proof on the `TestUnmanagedMiner` Miner State
+	tm.generatePreCommit(ctx, sectorNumber, preCommitSealRand, proofType, []abi.PieceInfo{})
+
+	// Step 4 : Submit the Pre-Commit to the network
+	r, err := tm.submitMessage(ctx, &miner14.PreCommitSectorBatchParams2{
+		Sectors: []miner14.SectorPreCommitInfo{{
+			Expiration:    2880 * 300,
+			SectorNumber:  sectorNumber,
+			SealProof:     TestSpt,
+			SealedCID:     tm.sealedCids[sectorNumber],
+			SealRandEpoch: preCommitSealRand,
+		}},
+	}, 1, builtin.MethodsMiner.PreCommitSectorBatch2)
+	req.NoError(err)
+	req.True(r.Receipt.ExitCode.IsSuccess())
+
+	// Step 5: Generate a ProveCommit for the CC sector
+	waitSeedRandomness := tm.proveCommitWaitSeed(ctx, sectorNumber)
+
+	proveCommit := tm.generateProveCommit(ctx, sectorNumber, proofType, waitSeedRandomness, []abi.PieceInfo{})
+
+	// Step 6: Submit the ProveCommit to the network
+	tm.t.Log("Submitting ProveCommitSector ...")
+
+	r, err = tm.submitMessage(ctx, &miner14.ProveCommitSectors3Params{
+		SectorActivations:        []miner14.SectorActivationManifest{{SectorNumber: sectorNumber}},
+		SectorProofs:             [][]byte{proveCommit},
+		RequireActivationSuccess: true,
+	}, 0, builtin.MethodsMiner.ProveCommitSectors3)
+	req.NoError(err)
+	req.True(r.Receipt.ExitCode.IsSuccess())
+
+	tm.proofType[sectorNumber] = proofType
+
+	respCh := make(chan WindowPostResp, 1)
+
+	go tm.wdPostLoop(ctx, sectorNumber, respCh, false)
+
+	return sectorNumber, respCh
+}
+
+func (tm *TestUnmanagedMiner) wdPostLoop(ctx context.Context, sectorNumber abi.SectorNumber, respCh chan WindowPostResp, withMockProofs bool) {
```

Comment:
this isn't a loop anymore, it just does one post then exits

which is going to be a problem because we're going to have additions (including ni-porep) that will take longer to onboard so existing sectors need to continue to post

which has implications for other changes in here -- the channel has a buffer of 1, but what if we just want to make it continue to post without waiting for a response? I think my original version had a channel for _first_ post because that's the only one we actually care about so we can check power.

It also disputed all posts to make sure it did the right thing on each post. I don't think that's as critical to keep in the loop, but it was a nice feature that you always got an assertion that your post was valid without having to include additional logic. The only catch with that is when you want to assert that a post _should not_ be valid, but that could also just be a boolean "expect valid" (to check each time and expect a failure to dispute) or "submit dispute" (to check each time or not check at all and make it the caller's responsibility).

I was just doing this last week with ni-porep, experimenting with a test case where the post was invalid and making sure the dispute didn't error to make sure that the dispute mechanism was working.

**However** I'm actually not sure about the value of adding tests for disputing succeeding, because we have itests/wdpost_dispute_test.go already. I don't know that ni-porep adds anything novel that would need to add a new dispute check on windowpost, we'd need to check on that. But I still think it's nice to add a dispute mechanism when we're not using fake proofs, even if we can't flip between "expect valid" and "expect invalid".